### DISCUSSION TITLE
Why are my group memberships not nested JSON

### DISCUSSION THREAD

**Discussion by kayx23** [keycloak/keycloak#28619] [Q&A]
I was trying to add custom groups to a client in a nested json structure.

I have configure token claim names to `acl_labels.nested.groups`:

![image](https://github.com/keycloak/keycloak/assets/39619599/3fec8c15-68a1-47cc-9ffb-e7168471f4e3)

The hoverover also says the dot will create a nested json structure:

![image](https://github.com/keycloak/keycloak/assets/39619599/8e0a0c70-84c1-434a-9373-90c88d6c6602)

However when I request the user info endpoint, the dots are literal dots in the field name:

![image](https://github.com/keycloak/keycloak/assets/39619599/a4347125-f192-4747-bedb-f35e54371c0a)

This is not consistent with my understanding of how it should work. How do I adjust my config so that it returns something like this?

```
{
  acl_labels: {
    nested: {
      groups: ["/group1", "/group2"]
    }
  }
}
```

Keycloak version: `18.0.2`

Thanks!

**Comment by kayx23** [comment]
Looks like this was a bug that was fixed in https://github.com/keycloak/keycloak/pull/20480. Trying to figure out which version I can upgrade my keycloak to.

**Reply by jonkoops** [reply]
You should upgrade to the latest version, all other versions are unsupported and have security issues.
